docs/api/cypress-api/ensure.mdx

```diff
@@ -0,0 +1,81 @@
+---
+title: Cypress.ensure
+---
+
+`Cypress.ensure.method()` is a collection of helper methods for verifying
+conditions while executing Cypress commands. They are mostly useful when writing
+[custom queries](/api/cypress-api/custom-queries) or
+[commands](/api/cypress-api/custom-commands).
+
+Most functions on `Cypress.ensure` accept `subject` argument, and assert
+something about it, throwing an error if it doesn't match. They have no return
+value; their only purpose is to ensure the state of the application and throw an
+error if it doesn't match.
+
+## Syntax
+
+```javascript
+// Command chain
+Cypress.ensure.isChildCommand(command, args, cy)​
+
+// Type of argument
+Cypress.ensure.isType(subject, type, name, cy)​
```

Comment:
I can't remember, but do these expect subject to be anything? We don't reference subject must in the docs and I'm thinking it might be helpful to expand above on what a subject might be for users who aren't familiar

Reply:
We reference 'subject' all over the docs.

https://docs.cypress.io/guides/core-concepts/introduction-to-cypress
> How Cypress manages subjects and chains of commands

The very first paragraph of the 'core concepts' guide discusses them.

> It's very important to understand the mechanism Cypress uses to chain commands together. It manages a Promise chain on your behalf, with each command yielding a 'subject' to the next command

https://docs.cypress.io/api/cypress-api/custom-commands#Arguments
> prevSubject | Boolean, String or Array | false | how to handle the previously yielded subject.
> In addition to controlling the command's implicit behavior you can also add declarative subject validations such as:
>    element: requires the previous subject be a DOM element
>    document: requires the previous subject be the document
>    window: requires the previous subject be the window

and all over that page.